src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/IUnconfiguredProjectCommonServices.cs

```diff
@@ -34,5 +34,10 @@ internal interface IUnconfiguredProjectCommonServices
         ///     Gets the <see cref="IProjectAccessor"/> which provides access to MSBuild evaluation and construction models for a project.
         /// </summary>
         IProjectAccessor ProjectAccessor { get; }
+
+        /// <summary>
+        /// Gets or sets if the project cache is applicable to this project.
+        /// </summary>
+        bool CacheApplicable { get; }
```

Comment:
It doesn't feel to me as though this member belongs here. This interface aggregates commonly used services, not arbitrary bits of state.

Instead I'd suggest:

- Create a new interface `IProjectSystemFeatureFlags` with `IsCacheApplicableAsync()` method
- Export an implementation that includes your changes from `UnconfiguredProjectCommonServices` and `FeatureFlagHelper`
- Import your new MEF part in places where you currently test the flag